From Olivier.Bornet@puck.ch  Thu Aug 20 14:33:11 2009
From: Olivier Bornet <Olivier.Bornet@puck.ch>
Date: Tue, 18 Aug 2009 21:05:57 +0200
Subject: USB: iuu_phoenix: add a way to select the default VCC
To: Greg KH <greg@kroah.com>
Cc: Alan Cox <alan@lxorguk.ukuu.org.uk>, linux-usb@vger.kernel.org, Olivier Bornet <Olivier.Bornet@puck.ch>
Message-ID: <1250622357-29765-6-git-send-email-Olivier.Bornet@puck.ch>


Using the module parameter vcc_default, you can choose the default VCC value.

Signed-off-by: Olivier Bornet <Olivier.Bornet@puck.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/serial/iuu_phoenix.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/iuu_phoenix.c
+++ b/drivers/usb/serial/iuu_phoenix.c
@@ -64,6 +64,7 @@ static int cdmode = 1;
 static int iuu_cardin;
 static int iuu_cardout;
 static int xmas;
+static int vcc_default = 5;
 
 static void read_rxcmd_callback(struct urb *urb);
 
@@ -115,7 +116,7 @@ static int iuu_startup(struct usb_serial
 		kfree(priv);
 		return -ENOMEM;
 	}
-	priv->vcc = 5;  /* 5 V for vcc by default */
+	priv->vcc = vcc_default;
 	spin_lock_init(&priv->lock);
 	init_waitqueue_head(&priv->delta_msr_wait);
 	usb_set_serial_port_data(serial->port[0], priv);
@@ -1339,3 +1340,7 @@ MODULE_PARM_DESC(clockmode, "Card clock 
 module_param(cdmode, int, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(cdmode, "Card detect mode (0=none, 1=CD, 2=!CD, 3=DSR, "
 		 "4=!DSR, 5=CTS, 6=!CTS, 7=RING, 8=!RING)");
+
+module_param(vcc_default, int, S_IRUGO | S_IWUSR);
+MODULE_PARM_DESC(vcc_default, "Set default VCC (either 3 for 3.3V or 5 "
+		"for 5V). Default to 5.");
